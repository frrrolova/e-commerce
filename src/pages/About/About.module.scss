@import '../../styles/variables.scss';

.outer {
  flex-direction: row;
  @media screen and (max-width: 850px){
    flex-direction: column;
  }
}

.inner {
  flex-direction: row;
  @media screen and (max-width: 850px){
    flex-direction: column;
  }
}

.logobox {
  width: 300px;
  min-width: 300px;
  @media screen and (max-width: 850px){
    width: 440px;
    min-width: 440px;
  }
  @media screen and (max-width: 470px){
    width: 300px;
    min-width: 300px;
  }
}

.logo {
  width: 284px;
  height: 284px;
  @media screen and (max-width: 850px){
    width: 420px;
    height: 420px;
  }
  @media screen and (max-width: 470px){
    width: 284px;
    height: 284px;
  }
}
