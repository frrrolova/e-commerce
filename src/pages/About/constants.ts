export const aboutConstants = {
  common: {
    title: `Our team`,
    subtitle: ` Made with love in 2024`,
  },
  collaboration: {
    title: `Collaboration`,
    p1: `We were working together as a group on a project for seven weaks. We created a project task board using a
        project management tool Trello to organize tasks into lists or columns corresponding to their status.`,
    p2: `We started each sprint with sprint planning, where we break down project tasks into smaller, manageable
        subtasks and assign them to each other, ensuring a clear delegation of responsibilities. During the
        sprint, we met for dailies, where each team member shared the status of their tasks. We shared workloads
        while discussing ideas, new methods, or different perspectives to attain better solutions. We actively
        participated in code reviews, providing constructive feedback and suggestions to our peers. At the end of
        the sprint we summed up the results and planned further work.`,
  },
};
