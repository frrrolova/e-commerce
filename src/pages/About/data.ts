import { TeamMember } from '@/types';
import tanykos from '/images/tanykos.webp';
import yar from '/images/Yar.webp';
import frrrolova from '/images/tatiana.webp';

export const members: TeamMember[] = [
  {
    role: 'Front-end developer',
    name: 'Tetiana Kostomitska',
    bio: `Graduated from Taurida National University with a degree in Mathematics. 
          Possesses 5 years of experience as a html-developer with expertise in web 
          development and responsive design. I enjoy learning new things and applying 
          knowledge in practice. I am happy to observe my progress in web-development.`,
    photo: tanykos,
    git: 'https://github.com/tanykos',
    contributions: [
      'CommerceTools Project Setup',
      'Trello Board Setup',
      'Main Page Implementation',
      'Header Implementation',
      'Catalog Page',
      'Routing Implementation',
      'Pagination',
    ],
  },
  {
    role: 'Team leader',
    name: 'Tatiana Frolova',
    bio: ` Graduated from St. Petersburg State Pediatric Medical University. Started learning frontend in 2022. Now live in Novi Sad, Serbia.
           In love with the process of creating interfaces, when you can visually observe the results of your work, I love the feeling of a working brain while solving a difficult task.`,
    photo: frrrolova,
    git: 'https://github.com/frrrolova',
    contributions: [
      'Environment Configuration',
      'Repository Setup',
      'Registration Page',
      'Routing Implementation',
      'User Profile Page',
      'Basket Implementation',
    ],
  },
  {
    role: 'Front-end developer',
    name: 'Yaroslaw Rychik',
    bio: 'My name is Yaroslaw. I am from Gomel. I have graduated from technical State University in 2019. Firstly I have worked as a design engineer but I made a decision to change my profession. So I went to an IT course at RS School. I hope one day I shall become a professional front-end developer.',
    photo: yar,
    git: 'https://github.com/REGEMLER',
    contributions: ['Development README', 'Login Page', 'Product Page', 'About Us Page'],
  },
];
