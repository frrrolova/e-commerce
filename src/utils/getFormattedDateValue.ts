export function getFormattedDateValue(value: string) {
  const date = new Date(value);
  const dateValue: string = new Date(date.getTime() - date.getTimezoneOffset() * 60000).toISOString().substring(0, 10);
  return dateValue;
}
