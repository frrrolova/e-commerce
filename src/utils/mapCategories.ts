import { Category } from '@/types';
import { Category as CategorySDK } from '@commercetools/platform-sdk';

export function mapCategories(categoryResponse: CategorySDK[]): Category[] {
  return categoryResponse.map((category) => ({
    id: category.id,
    name: category.name['en-US'],
    key: category.key,
    ancestors: category.ancestors,
    parent: category.parent,
  }));
}
