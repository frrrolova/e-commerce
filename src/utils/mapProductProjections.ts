import { Product } from '@/types';
import { ProductProjection } from '@commercetools/platform-sdk';

export function mapProductProjections(projections: ProductProjection[]): Product[] {
  return projections.map((projection) => ({
    id: projection.id,
    name: projection.name['en-GB'],
    description: projection.description?.['en-GB'] || 'Product has no description',
    images: projection.masterVariant.images,
    prices: projection.masterVariant.prices,
  }));
}
