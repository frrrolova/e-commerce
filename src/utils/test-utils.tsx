import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';

import type { AppStore, RootState } from '../store/store';
// As a basic setup, import your same slice reducers
import initStore from '../store/store';
import { MemoryRouter } from 'react-router-dom';
import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import { Project } from '@commercetools/platform-sdk';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

const mockShop: Project = {
  countries: [],
  createdAt: new Date().toISOString(),
  languages: [],
  key: 'test-key',
  version: 2,
  name: 'Plant Shop',
  currencies: [],
  messages: { enabled: false, deleteDaysAfterCreation: 15 },
  carts: {
    deleteDaysAfterLastModification: 90,
    countryTaxRateFallbackEnabled: false,
  },
};

export function renderWithProviders(ui: React.ReactElement, extendedRenderOptions: ExtendedRenderOptions = {}) {
  const {
    preloadedState = {
      project: mockShop,
    },
    // Automatically create a store instance if no store was passed in
    store = initStore(preloadedState.project as Project),
    ...renderOptions
  } = extendedRenderOptions;

  const Wrapper = ({ children }: PropsWithChildren) => (
    <MemoryRouter>
      <LocalizationProvider dateAdapter={AdapterDayjs}>
        <Provider store={store}>{children}</Provider>
      </LocalizationProvider>
    </MemoryRouter>
  );

  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
}
