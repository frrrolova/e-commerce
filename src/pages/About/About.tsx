import { Box, Container, Typography, Link, Paper } from '@mui/material';
import Logo from '/images/rs-logo.svg';
import TeamMemberCard from '@/components/TeamMemberCard/TeamMemberCard';
import { members } from './data';
import { aboutConstants } from './constants';

function About() {
  return (
    <>
      <Container
        data-testid={'about'}
        sx={{
          display: 'flex',
          flexDirection: 'column',
          gap: '20px',
          paddingX: '5px',
          paddingY: {
            xs: '10px',
            md: '30px',
          },
          width: '98%',
          margin: {
            xs: '30px auto',
            md: '60px auto',
          },
        }}
        component="main"
      >
        <Typography
          component="h1"
          sx={{
            width: '100%',
            fontSize: '26px',
            fontWeight: 'bold',
            color: '#3A7609',
            textAlign: 'center',
          }}
        >
          {aboutConstants.common.title}
        </Typography>
        <Box
          component="div"
          sx={{
            width: '100%',
            margin: '0 auto',
            display: 'flex',
            justifyContent: 'space-between',
            alignItems: 'center',
            gap: '20px',
            flexDirection: {
              xs: 'column',
              md: 'row',
            },
          }}
        >
          {members.map((item, index) => {
            return <TeamMemberCard key={index} teamMember={item} />;
          })}
        </Box>
        <Box
          component="div"
          sx={{
            width: '100%',
            display: 'flex',
            gap: '10px',
            flexDirection: {
              xs: 'column',
              md: 'row',
            },
          }}
        >
          <Box
            component="div"
            sx={{
              flexBasis: '70%',
              display: 'flex',
              flexDirection: 'column',
              gap: '10px',
            }}
          >
            <Typography
              component="h2"
              sx={{
                width: '100%',
                fontSize: '22px',
                fontWeight: 'bold',
                color: '#3A7609',
                textAlign: 'center',
              }}
            >
              {aboutConstants.collaboration.title}
            </Typography>
            <Typography
              component="p"
              sx={{
                textIndent: '20px',
              }}
            >
              {aboutConstants.collaboration.p1}
            </Typography>
            <Typography
              component="p"
              sx={{
                textIndent: '20px',
              }}
            >
              {aboutConstants.collaboration.p2}
            </Typography>
          </Box>
          <Box
            component="div"
            sx={{
              flexBasis: '30%',
              display: 'flex',
              flexDirection: 'column',
              alignItems: 'center',
              justifyContent: 'center',
            }}
          >
            <Paper
              elevation={3}
              sx={{
                overflow: 'hidden',
                backgroundColor: 'transparent',
                alignSelf: 'center',
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
                transition: 'background-color 0.3s ease-in-out',
                boxShadow: 'none',
                cursor: 'pointer',
                width: `300px`,
                minWidth: `300px`,
                padding: 1,
                borderRadius: '30px',
                '&:hover': { backgroundColor: '#323d2d' },
              }}
            >
              <Link href="https://rs.school/">
                <Box
                  component="img"
                  sx={{
                    objectFit: 'contain',
                    width: `${300 - 16}px`,
                    height: `${300 - 16}px`,
                  }}
                  src={Logo}
                  alt="RS School logo"
                />
              </Link>
            </Paper>

            <Typography
              sx={{
                p: 1,
              }}
              component="h2"
            >
              {aboutConstants.common.subtitle}
            </Typography>
          </Box>
        </Box>
      </Container>
    </>
  );
}

export default About;
