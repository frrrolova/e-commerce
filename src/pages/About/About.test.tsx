import { renderWithProviders } from '@/utils/test-utils';
import { screen, render } from '@testing-library/react';
import About from './About';

jest.mock('@client/client', () => {
  return null;
});

describe('About', () => {
  test('Data in document is correct', async () => {
    const { getByTestId } = renderWithProviders(<About />);

    expect(getByTestId('about')).toBeInTheDocument();
  });
  test('Title in the document is correct', async () => {
    render(<About />);
    const title = screen.getByText('Our team');

    expect(title).toBeInTheDocument();
  });
});
