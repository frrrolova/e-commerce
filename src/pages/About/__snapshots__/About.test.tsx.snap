// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Data in document is correct 1`] = `
<main
  class="MuiContainer-root MuiContainer-maxWidthLg css-17bh0pr-MuiContainer-root"
  data-testid="about"
>
  <div
    class="mainTitle MuiBox-root css-0"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-7uqf7p-MuiTypography-root"
    >
      Our team
    </h4>
  </div>
  <div
    class="MuiBox-root css-1bv4ar7"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-18v18mo-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-x8ntnm"
        >
          <header
            class="MuiBox-root css-hkmrg5"
          >
            <h2
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
              Tetiana Kostomitska
            </h2>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-136cz29-MuiTypography-root-MuiLink-root"
              href="https://github.com/tanykos"
            >
              Github
            </a>
          </header>
          <div
            class="MuiBox-root css-ig46ls"
          >
            <div
              class="MuiBox-root css-1szi7j8"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-1ncmvc0-MuiTypography-root"
            >
              Front-end developer
            </p>
            <h4
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Bio
            </h4>
            <h5
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
              Graduated from Taurida National University with a degree in Mathematics. 
          Possesses 5 years of experience as a html-developer with expertise in web 
          development and responsive design. I enjoy learning new things and applying 
          knowledge in practice. I am happy to observe my progress in web-development.
            </h5>
          </div>
          <footer
            class="MuiBox-root css-0"
          >
            <h3
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Contributions
            </h3>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                CommerceTools Project Setup
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Trello Board Setup
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Main Page Implementation
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Header Implementation
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Catalog Page
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Routing Implementation
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Pagination
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-18v18mo-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-x8ntnm"
        >
          <header
            class="MuiBox-root css-hkmrg5"
          >
            <h2
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
              Tatiana Frolova
            </h2>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-136cz29-MuiTypography-root-MuiLink-root"
              href="https://github.com/frrrolova"
            >
              Github
            </a>
          </header>
          <div
            class="MuiBox-root css-ig46ls"
          >
            <div
              class="MuiBox-root css-1szi7j8"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-1ncmvc0-MuiTypography-root"
            >
              Team leader
            </p>
            <h4
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Bio
            </h4>
            <h5
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
               Graduated from St. Petersburg State Pediatric Medical University. Started learning frontend in 2022. Now live in Novi Sad, Serbia.
           In love with the process of creating interfaces, when you can visually observe the results of your work. I love the feeling of a working brain while solving a difficult task.
            </h5>
          </div>
          <footer
            class="MuiBox-root css-0"
          >
            <h3
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Contributions
            </h3>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Environment Configuration
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Repository Setup
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Registration Page
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Routing Implementation
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                User Profile Page
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Basket Implementation
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-18v18mo-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-x8ntnm"
        >
          <header
            class="MuiBox-root css-hkmrg5"
          >
            <h2
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
              Yaroslaw Rychik
            </h2>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-136cz29-MuiTypography-root-MuiLink-root"
              href="https://github.com/REGEMLER"
            >
              Github
            </a>
          </header>
          <div
            class="MuiBox-root css-ig46ls"
          >
            <div
              class="MuiBox-root css-1szi7j8"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-1ncmvc0-MuiTypography-root"
            >
              Front-end developer
            </p>
            <h4
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Bio
            </h4>
            <h5
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
              My name is Yaroslaw. I am from Gomel. I have graduated from technical State University in 2019. Firstly I have worked as a design engineer but I made a decision to change my profession. So I went to an IT course at RS School. I hope one day I shall become a professional front-end developer.
            </h5>
          </div>
          <footer
            class="MuiBox-root css-0"
          >
            <h3
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Contributions
            </h3>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Development README
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Login Page
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                Product Page
              </div>
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                About Us Page
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1nnahni"
  >
    <div
      class="MuiBox-root css-1nxextx"
    >
      <div
        class="mainTitle MuiBox-root css-0"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-7uqf7p-MuiTypography-root"
        >
          Collaboration
        </h4>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-zvq0wx-MuiTypography-root"
      >
        We were working together as a group on a project for seven weaks. We created a project task board using a
        project management tool Trello to organize tasks into lists or columns corresponding to their status.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-zvq0wx-MuiTypography-root"
      >
        Our team effectively collaborated on the e-commerce store project by 
          maintaining open and constant communication through stand-up meetings. 
          Each member was assigned clear responsibilities, allowing for efficient task 
          management and accountability. We utilized tools like Trello, Diskord and 
          Telegram to track progress and address any issues in real-time. By 
          leveraging each member's expertise and bringing up a supportive 
          environment, we successfully developed a functional and user-friendly online plant store.
      </p>
    </div>
    <div
      class="MuiBox-root css-5ssijq"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-ne5q8u-MuiPaper-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
          href="https://rs.school/"
        >
          <img
            alt="RS School logo"
            class="MuiBox-root css-g98gbd"
            src="default"
          />
        </a>
      </div>
      <h2
        class="MuiTypography-root MuiTypography-body1 css-ym9mbz-MuiTypography-root"
      >
        Made with love in 2024
      </h2>
    </div>
  </div>
</main>
`;
